File: 22-07-2025_Smallest_positive_missing_number.py
class Solution:
    def missingNumber(self, arr):
        n = len(arr)

        # Place each number in its correct index if possible
        for i in range(n):
            while 1 <= arr[i] <= n and arr[i] != arr[arr[i] - 1]:
                arr[arr[i] - 1], arr[i] = arr[i], arr[arr[i] - 1]

        # Find the first index where the number is incorrect
        for i in range(n):
            if arr[i] != i + 1:
                return i + 1

        return n + 1

File: 22-07-2025_Smallest_positive_missing_number.cpp
class Solution {
  public:
    int missingNumber(vector<int> &arr) {
        // code here
        int n=arr.size();
        for(int i=0;i<n;i++){
                while(arr[i]>=1 && arr[i]<=n && arr[i]!=arr[arr[i]-1])
                    swap(arr[i],arr[arr[i]-1]);
            }
        for(int i=0;i<n;i++){
            if(i+1!=arr[i])return i+1;
        }
        return n+1;
    }
};